pandas/tests/groupby/test_groupby.py

```diff
@@ -2718,6 +2718,21 @@ def test_empty_dataframe_groupby(self):
 
         assert_frame_equal(result, expected)
 
+    def test_tuple_aggregation(self):
+        # Issue #18079
+        df = pd.DataFrame({'A': [1, 1, 1, 3, 3, 3],
+                           'B': [1, 1, 1, 4, 4, 4], 'C': [1, 1, 1, 3, 4, 4]})
+
```

Comment:
move to test_aggregate

can you parametrize with list-likes here (e.g. tuple/list,np.array,Series)

some of these might have different output (Series) but let's see.

you may have to have a tests with the parameterized ones and another one with 'other' things, e.g. 

```
def f(x):
    return tuple(x)
```

which work now.

Reply:
Thanks for the feedback, @jreback!  I'm hoping to tackle the list bug and these additional tests later this week.